crates/data_components/src/dynamodb/request_builder.rs

```diff
@@ -0,0 +1,1532 @@
+use crate::dynamodb::table_schema::DynamoDBTableSchema;
+use aws_sdk_dynamodb::types::AttributeValue;
+use datafusion::arrow::datatypes::SchemaRef;
+use datafusion::common::ScalarValue;
+use datafusion::common::tree_node::{TreeNode, TreeNodeRecursion};
+use datafusion::error::{DataFusionError, Result as DataFusionResult};
+use datafusion::logical_expr::{BinaryExpr, Expr, Operator};
+use derive_builder::Builder;
+use std::collections::{HashMap, HashSet};
+
+#[derive(Clone, Debug)]
+pub enum DynamoDBRequestPlan {
+    Query(QueryParams),
+    Scan(ScanParams),
+}
+
+#[derive(Builder, Debug, Default, Clone)]
```

Comment:
I do not like the derive builder, let's change this to a manually implemented one - similar to elsewhere in our codebase.

It is too much magic - just let AI generate the boilerplate.